### DISCUSSION TITLE
lag calculation in ForecasterAutoregMultiSeries

### DISCUSSION THREAD

**Discussion by mmajedi** [JoaquinAmatRodrigo/skforecast#762] [General]
I'm trying to figure out why lag_1 for level 0 on 2024-01-02 in example below is -1.56 as opposed to just 1. In-general how are lags calculated for the multi-series forecasters?

`import pandas as pd
date_range = pd.date_range(start='2024-01-01', end='2024-01-10')
item_1_values = range(1, 11)
item_2_values = range(11, 21)
df = pd.DataFrame({'item_1': item_1_values, 'item_2': item_2_values}, index=date_range)
forecaster = ForecasterAutoregMultiSeries(
                 regressor = RandomForestRegressor(random_state=123),
                 lags      = 1,
             )
x,y = forecaster.create_train_X_y(df)
x
`

**Comment by JavierEscobarOrtiz** [comment]
Hello @mmajedi 

This is because `ForecasterAutoregMultiSeries` uses a `StandardScaler()` (from scikit-learn) as the default transformer for all series (`transfomer_series` argument). Just set this argument to `None` when initializing the forecaster.

API Reference: https://skforecast.org/latest/api/forecastermultiseries
Scikit-learn transformers: https://skforecast.org/latest/user_guides/sklearn-transformers-and-pipeline

```Python
import pandas as pd 

date_range = pd.date_range(start='2024-01-01', end='2024-01-10') 
item_1_values = range(1, 11) 
item_2_values = range(11, 21) 
df = pd.DataFrame({'item_1': item_1_values, 'item_2': item_2_values}, index=date_range) 

forecaster = ForecasterAutoregMultiSeries(regressor = RandomForestRegressor(random_state=123), lags = 1, 
                                          transformer_series=None) 
x, y = forecaster.create_train_X_y(df) 
x
```
![imagen](https://github.com/user-attachments/assets/a5a863ab-d5f0-4eee-9720-975135d54004)

Hope it helps!

**Reply by mmajedi** [reply]
Thanks for your prompt reply Javier, I should have paid more attention to the API documentation. love the library btw
